app/src/main/java/org/thoughtcrime/securesms/conversation/v2/MessageDetailActivity.kt

```diff
@@ -0,0 +1,118 @@
+package org.thoughtcrime.securesms.conversation.v2
+
+import android.os.Bundle
+import android.view.View
+import android.widget.LinearLayout
+import androidx.annotation.DimenRes
+import kotlinx.android.synthetic.main.activity_conversation_v2_action_bar.*
+import kotlinx.android.synthetic.main.activity_message_detail.*
+import network.loki.messenger.R
+import org.session.libsession.messaging.MessagingModuleConfiguration
+import org.session.libsession.messaging.messages.visible.LinkPreview
+import org.session.libsession.messaging.messages.visible.OpenGroupInvitation
+import org.session.libsession.messaging.messages.visible.Quote
+import org.session.libsession.messaging.messages.visible.VisibleMessage
+import org.session.libsession.messaging.sending_receiving.MessageSender
+import org.session.libsession.messaging.utilities.UpdateMessageData
+import org.session.libsession.utilities.Address
+import org.session.libsession.utilities.ExpirationUtil
+import org.session.libsession.utilities.TextSecurePreferences
+import org.session.libsession.utilities.recipients.Recipient
+import org.thoughtcrime.securesms.PassphraseRequiredActionBarActivity
+import org.thoughtcrime.securesms.database.DatabaseFactory
+import org.thoughtcrime.securesms.database.model.MessageRecord
+import org.thoughtcrime.securesms.database.model.MmsMessageRecord
+import org.thoughtcrime.securesms.util.DateUtils
+import java.text.SimpleDateFormat
+import java.util.*
+import kotlin.math.roundToInt
+
+
+class MessageDetailActivity: PassphraseRequiredActionBarActivity() {
+
+    var messageRecord: MessageRecord? = null
+
+    // region Settings
+    companion object {
+        // Extras
+        const val MESSAGE_TIMESTAMP = "message_timestamp"
+    }
+    // endregion
+
+    override fun onCreate(savedInstanceState: Bundle?, ready: Boolean) {
+        super.onCreate(savedInstanceState, ready)
+        setContentView(R.layout.activity_message_detail)
+        title = resources.getString(R.string.conversation_context__menu_message_details)
+        val timestamp = intent.getLongExtra(MESSAGE_TIMESTAMP, -1L)
+        val author = Address.fromSerialized(TextSecurePreferences.getLocalNumber(this)!!)
```

Comment:
Would be nice to add a comment here explaining that this is only true because we currently only show this screen if an outgoing message failed to send.